tools/tpm_keystore/cert_encrypt_tpm.c

```diff
@@ -0,0 +1,500 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information.
+ *
+ * Copyright (c) 2021 Kalycito Infotech Private Limited
+ */
+
+/* gcc cert_encrypt_tpm.c -o cert_encrypt_tpm -ltpm2_pkcs11 -lssl -lcrypto */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <openssl/evp.h>
+
+#include "pkcs11.h"
+
+typedef enum { B_FALSE, B_TRUE } boolean_t;
+
+typedef struct binary_data binary_data;
+struct binary_data {
+  long length;
+  void *data;
+};
+
+#define ARRAY_LEN(x) (sizeof(x) / sizeof(x[0]))
+
+CK_RV encrypt(int slotNum, unsigned char *pin, unsigned char *label, binary_data *in_data, binary_data **encrypted_data, binary_data * iv_data);
+boolean_t find_object_by_label(CK_SESSION_HANDLE hSession, unsigned char *label, CK_OBJECT_HANDLE *object_handle);
+static void process_command_line(int argc, char** argv, unsigned char** slot, unsigned char** pin, unsigned char** label, unsigned char** in_file, unsigned char** out_file);
+binary_data* read_input_file(const char * filename);
+boolean_t write_output_file(const char * filename, binary_data* data);
+void get_MAC(const uint8_t *message, size_t message_len, unsigned char **message_digest, unsigned int *message_digest_len);
+void get_MAC_error();
+
+int main(int argc, char** argv)
```

Comment:
Find out which capabilities you need from the TPM and check their existence before using the TPM. Print a meaningful error message listing the missing capabilities.